test/worker_test.py

```diff
@@ -2135,3 +2136,61 @@ def run(self):
                 self.assertEqual(0, self.sch._state.get_task(s1.task_id).failures.num_failures())
                 self.assertEqual(self.per_task_retry_count, self.sch._state.get_task(e2.task_id).failures.num_failures())
                 self.assertEqual(self.default_retry_count, self.sch._state.get_task(e1.task_id).failures.num_failures())
+
+
+class SleepyLocalTarget(luigi.LocalTarget):
+    time_to_sleep = 0.5
+
+    def exists(self):
+        time.sleep(self.time_to_sleep)
```

Comment:
I'm a bit worried about this. We reserve the right to check `complete()` multiple times in luigi, and if an "refactoring" changes this (checks more often or for example caches and checks less), then this test-case might start to break.

Reply:
The test checks that the `Parent` task won't be run twice, so it shouldn't fail even if we add more or less `complete()` checks in the future